.circleci/config.yml

```diff
@@ -12,93 +12,126 @@
 
 version: 2.1
 orbs:
-  android: circleci/android@2.1.2
+  android: circleci/android@2.4.0
+  codecov: codecov/codecov@4.0.1
 
 # Workflows orchestrate a set of jobs to be run;
 workflows:
   version: 2
   build-test-deploy:
     jobs:
-      - build-and-unit-test
-      - functional-test
+      - validate-code
+      - build-and-unit-test:
+          requires:
+            - validate-code
+      - functional-test:
+          requires:
+            - validate-code
  
 jobs:
+  validate-code:
+    executor:
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1-node
+
+    steps:
+      - checkout
+
+      - android/restore-gradle-cache
+
+      - android/restore-build-cache
+
+      # - run:
+      #     name: Check Format
+      #     command: make checkformat
+
+      # - run:
+      #     name: Check Style
+      #     command: make checkstyle
+
+      - android/save-gradle-cache
+
+      - android/save-build-cache
+
+      # Store Lint report
```

Comment:
we can remove the store link report comment.

Reply:
yeap 👍🏼  will do